src/content/docs/tutorials/crates-tui/app_handle_event.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Handle Events
+---
+
+In the `run` method, we have the following code:
+
+```rust
+if let Some(e) = events.next().await {
+    self.handle_event(e)?.map(|action| self.tx.send(action));
+}
+```
+
+`events.next().await` returns an `Event` from the stream we implemented in the previous section.
+This `event` is passed to the `handle_event` method.
+
+`handle_event` returns an `Option<Action>` which we map to an `Action` that has to be acted on by
+the app:
+
+```rust
+{{#include @code/crates-tui-tutorial-app/src/app.rs:app_handle_event}}
+```
+
+Most of the work in deciding which `Action` should be taken is done in `handle_key_event`.
```

Comment:
It should be widget specific. Defining keys at the application level is too central.

Reply:
I don't think it should be widget specific but rather mode specific. Happy to move it as part of a method on `Mode`.

Reply:
Sounds reasonable. I say widget, but meant something aligned with the vertical slice that is meaningful. Mode might be a good enough compromise on that.

Reply:
done in recent commit.